include/deal.II/base/discrete_time.h

```diff
@@ -71,19 +83,43 @@ class DiscreteTime
   /**
    * Return the end of the time interval.
    * The final time step ends exactly at this point. This exact floating-point
-   * equality is very important because it allows us to use the expression
-   * <code>time.get_current_time() != time.get_end_time()</code> as the
-   * conditional statement in a for loop to check if the end time is reached.
+   * equality is very important because it allows us to equality-compare
+   * current time with end time and decide whether we have reached the end of
+   * the simulation.
    */
   double
   get_end_time() const;
 
+  /**
+   * Return whether no step has taken place yet.
+   */
+  bool
+  is_at_start() const;
+
+  /**
+   * Return whether time has reached the end time.
+   */
+  bool
+  is_at_end() const;
+
   /**
    * Return the size of the step from current time step to the next.
    */
   double
   get_next_step_size() const;
 
+  /**
+   * Return the step size of the previous step.
+   */
+  double
+  get_previous_step_size() const;
```

Comment:
I cannot decide for myself if this is the correct name for this function. Let me propose a situation, and maybe someone could help me think through it. 

Hypothetically, we might want to support a situation where we've chosen too large a timestep size and want to cut back to some other interval between the previous (accepted) time and the current time. I'd argue that the current time has therefore been proposed, and is not concrete until we choose to iterate / advance in time. In this case, does it make sense that we'd be modifying the "previous_step_size"? To me, something in the past is immutable. I'd therefore argue that this should be called `get_current_step_size()`.

Maybe what I've mentioned above is not philosophically correct, and that we're actually at a set time ("current") and trying to move forward to the next time. Then what step I choose is certainly mutable, as it has not been "accepted" into the discrete time history. In this case we'd almost certainly be asking what the "current step size" is to move from the current time to the future (to be accepted) time.

Reply:
Correction: What I've stated above would be most applicable to time incrementation strategies commonly adopted for implicit methods. Step-21 is an explicit method so `current_step_size` is actually the `next_step_size`.